core/java/com/google/instrumentation/stats/MeasurementDescriptor.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.stats;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * MeasurementDescriptor.
+ *
+ * <p>Note: MeasurementDescriptor names are {@link String}s with enforced restrictions.
+ */
+public final class MeasurementDescriptor {
+  public static final int MAX_LENGTH = StringUtil.MAX_LENGTH;
+
+  /**
+   * Name of measurement, e.g. rpc_latency, cpu. Must be unique.
+   */
+  public final String name;
+
+  /**
+   * Detailed description of the measurement, used in documentation.
+   */
+  public final String description;
+
+  /**
+   * The units in which {@link MeasurementDescriptor} values are measured.
+   */
+  public final MeasurementUnit unit;
```

Comment:
personally, I prefer the style here (except not with an exposed public ctor). It is simpler and still allows new fields. Representations can change without changing the canonical struct (Ex we do the same thing in zipkin)